shed/tests/test_streams.py

```diff
@@ -454,6 +454,25 @@ def f(img1):
         assert n in assert_docs
 
 
+def test_filter_full_header(exp_db, start_uid1):
+    source = Stream()
+
+    def f(docs):
+        d = docs[0]
+        return d['sample_name'] != 'hi'
```

Comment:
yeah, maybe AB testing or something (I forget what they call it). simulate the input in a well defined way without depending on the object itself. could be a good idea.